pages/report.tsx

```diff
@@ -0,0 +1,300 @@
+import React, { useEffect, useState } from 'react';
+import TextArea from 'antd/lib/input/TextArea';
+import Link from 'next/link';
+import { RequestType, useCustomSWR } from '../utils/request-utils';
+import { GitHubIssueStatus, Report } from '../types/client';
+import { ColumnsType } from 'antd/lib/table';
+import { Octokit } from 'octokit';
+import { useRouter } from 'next/router';
+
+const ReportBug = () => {
+	const { data: session, status } = useSession();
+	const router = useRouter();
+
+	const [success, setSuccess] = useState<boolean | null>(null);
+	const [loading, setLoading] = useState<boolean>(false);
+	const [githubIssues, setGithubIssues] = useState<GitHubIssueStatus[] | null>(null);
+
+	useEffect(() => {
+		const fetchGithubIssue = async () => {
+			const octokit = new Octokit({
+				auth: process.env.GITHUB_TOKEN,
+			});
+
+			await octokit
+				.request('GET /repos/VandyHacks/witness/issues?state=all', {
+					owner: 'VandyHacks',
+					repo: 'witness',
+					headers: {
+						'X-GitHub-Api-Version': '2022-11-28',
+					},
+				})
+				.then(res => {
+					// parse through issues and match the issue number and status
+					console.log(res.data);
+					const issues = res.data.map((issue: any) => {
+						return {
+							issueNumber: issue.number,
+							status: issue.state,
+						};
+					});
+					setGithubIssues(issues);
+				})
+				.catch(err => {
+					console.error(err);
+				});
+		};
+
+		fetchGithubIssue();
+	}, []);
+
+	const handleSubmitBug = async (input: any) => {
+		setLoading(true);
+		const newReport = {
+			email: session?.user?.email,
+			name: session?.user?.name,
+			role: session?.userType || 'HACKER',
+			date: new Date().toISOString(),
+			description: input.description,
+		};
+		const res = await fetch('/api/report', {
+			method: 'POST',
+			headers: {
+				'Content-Type': 'application/json',
+			},
+			body: JSON.stringify(newReport),
+		});
+
+		if (res.status === 200) {
+			setSuccess(true);
+		} else {
+			setSuccess(false);
+		}
+		setLoading(false);
+	};
+
+	const { data: bugReports, error: bugReportsError } = useCustomSWR<Report[]>({
+		url: '/api/report',
+		method: RequestType.GET,
+		errorMessage: 'Failed to get bug reports.',
+	});
+
+	if (!session && status === 'unauthenticated') router.push('/');
+
+	const newCols: ColumnsType<Report> = [
+		{
+			title: 'Description',
+			dataIndex: 'description',
+			key: 'description',
+			width: '40%',
+		},
+		{
+			title: 'Date',
+			dataIndex: 'date',
+			key: 'date',
+			width: '20%',
+			render: (date: string) => {
+				return <div>{new Date(date).toLocaleString()}</div>;
+			},
+		},
+		{
+			title: 'Status',
+			dataIndex: 'status',
+			key: 'status',
+			width: '20%',
+			render: (_: string, record: Report) => {
+				return (
+					<Tag
+						color={
+							githubIssues?.find((issue: any) => issue.issueNumber === record.ghIssueNumber)?.status ===
+							'open'
```

Comment:
these lines are used in multiple places. it's hard to grasp what it does at a glance. 
we should abstract this into a function (with a simple comment) and call that instead.